CHANGES.rst

```diff
@@ -6,6 +6,13 @@
 
 - Add support for Python 3.7 and drop support for Python 3.3.
 
+- Build the CFFI modules (used on PyPy or when PURE_PYTHON is set) at
+  installation or wheel building time when CFFI is available. This
+  replaces the deprecated way of building them at import time. If
+  binary wheels are distributed, it eliminates the need to have a
+  functioning C compiler to use PyPy. See `issue 75
+  <https://github.com/zopefoundation/persistent/issues/75>`_.
```

Comment:
My CFFIfu is weak: can you add a link to docs about how the "modern way" is supposed to work?

Reply:
Sure will. It's https://cffi.readthedocs.io/en/latest/cdef.html#ffibuilder-set-source-preparing-out-of-line-modules for "API + out of line"

What we *were* doing was "API + in line", [which is deprecated](https://cffi.readthedocs.io/en/latest/overview.html#abi-versus-api):

> Note that the “API level + in-line” mode combination exists but is long deprecated. It used to be done with lib = ffi.verify("C header"). The out-of-line variant with set_source("modname", "C header") is preferred.

Reply:
Wonderful, thanks!